src/common/dimension.hh

```diff
@@ -0,0 +1,26 @@
+#ifndef H_ACQUIRE_STORAGE_ZARR_COMMON_DIMENSION_V0
+#define H_ACQUIRE_STORAGE_ZARR_COMMON_DIMENSION_V0
+
+#include "device/props/storage.h"
+
+#include <string>
+
+namespace acquire::sink::zarr {
+struct Dimension
+{
+  public:
+    explicit Dimension(const std::string& name,
```

Comment:
> The caller can initialize the object this way:
>
> ```
> const auto d = Dimension {{
>   .name = "name",
>   .kind = ...,
>   .array_size_pix = ...
>   ...
> }};
> ```

Not exactly. `StorageDimension::name` is a `struct String` which has the following interface:

```c
struct String
{
    char* str;
    size_t nbytes;
    uint8_t is_ref;
};
```

So there's a little more manual intervention than your example.

Anyway, there are two constructors because there are two modes of initialization. The first is from a `struct StorageDimension` in `Zarr::set_dimensions_()`, and the second is in `downsample(...)` in `writer.cpp` (line 132), taking a `Dimension` and downsampling it.